### PR TITLE
test(app-project): mock Zooniverse APIs in stories with MSW

### PR DESCRIPTION
**Pull request by eatyourgreens**
- install MSW for the storybook.
- add a mock API to the workflow selector stories.
- add mock APIs to the subject picker stories.
- ~~skip a couple of Mocha tests that now fail with errors.~~ all the tests should still pass.

## Package
- app-project

## Linked Issue and/or Talk Post
- companion to #6219.
- adds a story to test #6198.

## How to Review
Running the dev storybook locally, there are a couple of new stories:
- the subject picker now shows mock data in the browser!
http://localhost:9001/?path=/story/project-app-shared-subject-picker--default
A story showing the subject picker component with subjects loaded in from a mock API.

- the workflow selector has a story that tests the case where an assigned workflow is complete, and is fetched from the Panoptes API (#6198.) It unlocks levels 1 and 2, even though 3 is complete and not available. Level 4 remains locked. 
http://localhost:9001/?path=/story/project-app-shared-workflow-selector--missing-levels
A story showing locked and unlocked workflow levels, when the assigned workflow is complete.

You can test the deployed storybook (`NODE_ENV=production`) with:

```sh
yarn build-storybook
npx serve ./storybook-static
```
~~I've skipped a couple of tests that are failing with errors. I'm not sure why they fail. I think it might be happening because stories are designed to run in a browser, but the tests run in Node.~~ Mocha tests, with `nock` to mock the Panoptes API in Node, should still run and pass on this branch.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

### CODE REVIEW

File: packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js [hunk 1]
```diff
@@ -92,11 +36,11 @@ describe('Components > Subject Picker', function () {
     expect(columnHeadings).to.exist()
   })
 
-  it('should have a row for each subject', function () {
+  it.skip('should have a row for each subject', function () {
```
Comment (eatyourgreens) [inline packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js:39]: The mock request that populates the table doesn't seem to be running in JSDOM. It does run when the story is viewed in a browser.

File: packages/app-project/.storybook/preview.js
Comment (eatyourgreens) [inline packages/app-project/.storybook/preview.js:1]: I've used the same config here that I use to test client-side React componnents with Storybook 8, so this should be forward compatible.

File: packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js]: Turns out that requests using `superagent` still need `nock` in Node. I'm not sure why.

File: packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js [hunk 2]
```diff
@@ -1,87 +1,42 @@
...
+      .query(true)
+      .reply(200, {
+        subjects: [
+          { id: 1, already_seen: false, retired: false },
+          { id: 2, already_seen: true, retired: false },
+          { id: 3, already_seen: true, retired: true }
+        ]
+      })
+    const DefaultStory = composeStory(Default, Meta)
+    await applyRequestHandlers(DefaultStory.parameters.msw)
```
Comment (eatyourgreens) [inline packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js:24]: This is needed in Storybook 7, but 8 handles it automatically for you.

File: packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.spec.js]: Anyhow, this test isn't skipped any more, and is working as expected.

**Comment by goplayoutside3** [comment]
Phew 😌 I messed up there for a minute by pushing changes before including the latest dependency bumps from the master branch.

I made two changes:
1) Per https://github.com/zooniverse/front-end-monorepo/pull/6219#issuecomment-2474086731 I changed the SWROptions in `useAssignedLevel()` to revalidate on mount. Previously, the WorkflowSelector story on this branch did not properly unlock Levels 1+2 when the user is assigned a workflow id of an inactive workflow with level 3.
2) I made some updates to the WorkflowSelector stories to use a select with options for `userReadyState` (its `options` array was incorrectly configured) and I renamed the two stories that use the workflow assignment feature to reflect the name of that feature.

**Review by goplayoutside3 — APPROVED** [review]

**Comment by goplayoutside3** [comment]
I think this change may have resulted in a failed Storybook deploy. I haven't had a chance to look into it yet, but the links to the action is: https://github.com/zooniverse/front-end-monorepo/actions/runs/11956712268/job/33332112305

**Comment by eatyourgreens** [comment]
The action is failing in the `lib-classifier` storybook build, but the log doesn’t give much more info than that.

**Comment by eatyourgreens** [comment]
It failed on this deploy too, I think maybe before this PR merged. 

https://github.com/zooniverse/front-end-monorepo/actions/runs/11822175353/job/32938615231

Weird, because CI builds all the storybooks for every PR.

**Comment by eatyourgreens** [comment]
The last working deploy was this commit, if that helps.
https://github.com/zooniverse/front-end-monorepo/commit/860f09fc57890bdc6a4f0fb7706b30c8ab01d186

**Pull request merged**